src/funcCoreTools/funcHostTask.ts

```diff
@@ -86,13 +90,32 @@ export function isFuncHostTask(task: vscode.Task): boolean {
     return /func (host )?start/i.test(commandLine || '');
 }
 
+let latestTerminalShellExecutionEvent: vscode.TerminalShellExecutionStartEvent | undefined;
 export function registerFuncHostTaskEvents(): void {
     registerEvent('azureFunctions.onDidStartTask', vscode.tasks.onDidStartTaskProcess, async (context: IActionContext, e: vscode.TaskProcessStartEvent) => {
+        const terminalEventReader = vscode.window.onDidStartTerminalShellExecution(async (terminalShellExecEvent) => {
```

Comment:
Had some questions about this part, reaching out offline

Reply:
Talked offline; registering the event handler outside of the `onDidStartTaskProcess`